src/vs/workbench/contrib/debug/browser/debugConfigurationManager.ts

```diff
@@ -715,10 +717,29 @@ class WorkspaceLaunch extends AbstractLaunch implements ILaunch {
 		return this.configurationService.inspect<IGlobalConfig>('launch').workspaceValue;
 	}
 
-	async openConfigFile(sideBySide: boolean, preserveFocus: boolean): Promise<{ editor: IEditorPane | null, created: boolean }> {
+	async openConfigFile(sideBySide: boolean, preserveFocus: boolean, type?: string, token?: CancellationToken): Promise<{ editor: IEditorPane | null, created: boolean }> {
+		const ws = this.contextService.getWorkspace();
+
+		if (ws.folders.length === 0) {
+			return { editor: null, created: false };
+		}
+
+		let launchExistInFile = !!this.getConfig();
+		if (!launchExistInFile) {
+			// Launch property in workspace config not found: create one by collecting launch configs from debugConfigProviders
+			let content = '';
+			const adapter = await this.configurationManager.guessDebugger(type);
```

Comment:
I see we have similar code in the `Launch`. Might it make sense to extract this code to the `AbstractLaunch` so both clases use it? Not sure if 100% good idea..